Support added for RPi4

DesobedienteTecnologico opened the pull request:
Configuration added for RPi4:

✅ Due is not necessary to have a custom kernel in dev image, I removed it and pointed directly to the default kernel configuration.
⬜ Mount/Umount MicroSD (**WIP**)
⬜ Add pi4 build option to build.sh script

---

### Some significant changes from others versions:

Raspberry Pi4 firmware variant added:

**config_seedsigner.txt**
- start_file=start**4**x.elf
- fixup_file=fixup**4**x.dat



**genimage-rpi-seedsigner.cfg**
- "rpi-firmware/fixup**4**x.dat",
- "rpi-firmware/start**4**x.elf",



**pi4_defconfig / pi4-dev_defconfig**
- BR2_PACKAGE_RPI_FIRMWARE_VARIANT_**PI4**_X=y



Then of course custom Kernel configuration file was added with changes that belong to the new architecture and RPi firmware.

DesobedienteTecnologico committed:
Support added for RPi4

Configuration added for RPi4:

1. Due is not necessary to have a custom kernel in dev image, I removed it and pointed directly to the default kernel configuration.
2. [ ] Mount/Umount MicroSD (WIP)

newtonick commented:
@DesobedienteTecnologico the microsd insert/remove notification works on pi4-dev config. Just tested it by pulling this PR locally and building using the config in pi4-dev.

newtonick commented:
TODO item for this PR:

- [ ] Add pi4 build option to build.sh script

DesobedienteTecnologico commented:
> @DesobedienteTecnologico the microsd insert/remove notification works on pi4-dev config. Just tested it by pulling this PR locally and building using the config in pi4-dev.

Probably there is the need to add more Kernel modules on it.
I will check ASAP. Thanks for the feedback.

DesobedienteTecnologico committed:
Merge pull request #23 from newtonick/restructure

Restructure and add Docker from DT repo

newtonick committed:
first pass at improving readme

newtonick committed:
minor cleanup

newtonick committed:
fix indentation

newtonick committed:
add development cycle description and examples

newtonick committed:
dev configs

newtonick committed:
remove gzip of img file and remove the move command of the config folder

newtonick committed:
Add skip-build option to use with  keep-alive

newtonick committed:
cleanup

DesobedienteTecnologico committed:
Merge pull request #26 from newtonick/remove-gzip

Remove gzip of img file and remove the move command of the config folder

DesobedienteTecnologico committed:
Merge pull request #27 from newtonick/add-skip-build-option

Add skip-build option to use with keep-alive

DesobedienteTecnologico committed:
libncurses-dev added in Dockerfile

libncurses-dev is required to use `make menuconfig` and others menus.

DesobedienteTecnologico committed:
Pi4 added to build script

newtonick committed:
Update build.sh to default to 0.5.2 release

newtonick committed:
updated to add additional note from DT (https://desobedientetecnologico.gitlab.io/blog/posts/seedsigner_os/)

DesobedienteTecnologico committed:
Merge pull request #24 from newtonick/update-readme-202212

LGTM

DesobedienteTecnologico committed:
Merge pull request #33 from newtonick/patch-7

The version change LGTM
Update build.sh to default to 0.5.2 release

newtonick committed:
update embit to version 0.7.0 adding the required bip-85 support needed for seedsigner app repo

SeedSigner committed:
Merge pull request #30 from DesobedienteTecnologico/libncurses-lib_dockerfile

libncurses-dev added in Dockerfile

DesobedienteTecnologico committed:
Merge pull request #35 from newtonick/update-embit-0.7.0

LGTM

Update embit to version 0.7.0

DesobedienteTecnologico committed:
Fix - umount MicroSD

DesobedienteTecnologico committed:
Merge branch 'main' into rpi4_support

DesobedienteTecnologico committed:
File mode fixed

- Unexpected file mode change to 644 in last commit. Now is 755 again.
- Pi2 flag added. unintentionally deleted.

DesobedienteTecnologico committed:
Camera Kernel modules added and bootcode.bin removed

- Camera Kernel modules added.
- bootcode.bin remove. RPi4 do not need it to boot. Bootloader is on EEPROM.
Reference:
https://www.raspberrypi.com/documentation/computers/raspberry-pi.html#raspberry-pi-4-boot-flow
